### ISSUE TITLE
Memory requirements for ContraCode

### ISSUE THREAD

**Issue by parasj** [parasj/contracode#17]
Hi Parasj, thanks for publishing your code. I want to ask what are the parameters of the equipment used in your experiment? I found that 16G memory is not enough if using the javascript_augmented.pickle.gz file.

_Originally posted by @QZH-eng in https://github.com/parasj/contracode/issues/6#issuecomment-726468747_

**Comment by parasj** [comment]
*Copying from #6*

Pretraining is memory hungry as contrastive learning benefits from large batch sizes (see https://arxiv.org/abs/2002.05709). Moreover, the transformer backbone we leverage uses significantly more memory than typical image classification architectures.

We generally performed pretraining over 2-4 16GB V100 GPUs. We provide pretrained checkpoints due to the large cost of pretraining. Finetuning is very cheap and was performed on 1 V100 GPU.

Some recommendations to reduce memory consumption:
(1) reducing the sequence length for the Transformer encoder
(2) decreasing the hidden dimension size of our model
(3) adding checkpoint annotations for gradient checkpointing (e.g. PyTorch gradient checkpointing)

**Comment by QZH-eng** [comment]
Thanks for your reply.
Your idea is very novel,and I will try it on your advice.
Thanks again.

**Comment by parasj** [comment]
Thanks! Feel free to reopen the issue if you have any further questions.
